extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -912,6 +938,48 @@ export class Bind {
   }
 
   /**
+   * Deep merge object b into object a. Both a and b can only contain
+   * primitives, arrays, and objects created by Object.create(null).
+   * Arrays are replaced, not merged. Other objects are recursively merged.
+   * @param {!Object} a the destination object
+   * @param {!Object} b
+   * @param {number=} opt_depth The depth of the objects being merged
+   * @return {!Object}
+   */
+  deepMerge_(a, b, opt_depth) {
```

Comment:
This may also be a good place to check for and ignore invalid (non-JSON) objects... my TODO on line 172. 😄